components/org.wso2.carbon.identity.application.authenticator.hypr.common/src/main/java/org/wso2/carbon/identity/application/authenticator/hypr/common/web/HYPRAuthorizationAPIClient.java

```diff
@@ -0,0 +1,207 @@
+     *                                  registered devices via the api call
+     */
+    public static RegisteredDevicesResponse getRegisteredDevicesRequest(String baseUrl, String appId, String apiToken,
+                                                                        String username)
+            throws HYPRAuthnFailedException {
+
+        // Device info URL : {{baseUrl}}/rp/ api/oob/client/authentication/{{appId}}/{{username}}/devices
+        String deviceInfoURL = String.format("%s%s%s/%s/devices", baseUrl,
+                HyprAuthenticatorConstants.HYPR.HYPR_USER_DEVICE_INFO_PATH, appId, username);
+
+        try {
+            HttpResponse response = HYPRWebUtils.httpGet(apiToken, deviceInfoURL);
+
+            if (response.getStatusLine().getStatusCode() == 401) {
+                throw getHyprAuthnFailedException(
+                        HyprAuthenticatorConstants.ErrorMessages.HYPR_ENDPOINT_API_TOKEN_INVALID_FAILURE);
+            } else if (response.getStatusLine().getStatusCode() > 400) {
+                throw getHyprAuthnFailedException(
+                        HyprAuthenticatorConstants.ErrorMessages.AUTHENTICATION_FAILED_RETRIEVING_REG_DEVICES_FAILURE);
+            }
+
+            Gson gson = new GsonBuilder().create();
+            HttpEntity entity = response.getEntity();
+            String jsonString = EntityUtils.toString(entity);
+            Type listType = new TypeToken<List<RegisteredDevice>>() {
+            }.getType();
+
+            return new RegisteredDevicesResponse(gson.fromJson(jsonString, listType));
+
+        } catch (IOException e) {
+            throw getHyprAuthnFailedException(HyprAuthenticatorConstants.ErrorMessages
+                    .AUTHENTICATION_FAILED_RETRIEVING_REG_DEVICES_FAILURE, e);
+        } catch (HYPRClientException e) {
+            throw getHyprAuthnFailedException(HyprAuthenticatorConstants.ErrorMessages
+                    .SERVER_ERROR_CREATING_HTTP_CLIENT, e);
+        }
+    }
+
+    /**
+     * Call the HYPR server API to initiate the authentication request via sending the push notification to the
+     * registered device.
+     *
+     * @param baseUrl   The baseURL provided from the HYPR.
+     * @param appId     The ID of the application created via the HYPR control center.
+     * @param apiToken  The API token generated from the application created via the HYPR control center.
+     * @param username  The username provided by the user.
+     * @param machineId The machineId unique per user.
+     * @return DeviceAuthenticationResponse
+     * @throws HYPRAuthnFailedException Exception throws when there is an error occurred when initiating the
+     *                                  authentication with HYPR server
+     */
+    public static DeviceAuthenticationResponse initiateAuthenticationRequest
+    (String baseUrl, String appId, String apiToken, String username, String machineId)
```

Comment:
Let's move the open bracket to the first line

Reply:
Fixed with [eb39195](https://github.com/wso2-extensions/identity-outbound-auth-hypr/pull/1/commits/eb39195b388a647223846891193117538a081295)